test/components/groupTests.ts

```diff
@@ -1,210 +1,208 @@
+      cg2.renderTo(svg);
 
-    c1Node = svg.select(".component-1").node();
-    c2Node = svg.select(".comopnent-2").node();
+      assert.strictEqual(cg2.components().length, 0,
+        "second group should have no component before movement");
 
-    assert.isNotNull(c1Node, "component 1 is still in the DOM");
-    assert.isNull(c2Node, "component 2 was removed from the DOM");
+      assert.strictEqual(cg1.components().length, 1,
+        "first group should have 1 component before movement");
 
-    cg.detach();
-    let cgNode = svg.select(".component-group").node();
-    c1Node = svg.select(".component-1").node();
+      assert.strictEqual(c.parent(), cg1,
+        "component's parent before moving should be the group 1"
+      );
 
-    assert.isNull(cgNode, "component group was removed from the DOM");
-    assert.isNull(c1Node, "componet 1 was also removed from the DOM");
+      assert.doesNotThrow(() => cg2.append(c), Error,
+        "should be able to move components between groups after anchoring"
+      );
 
-    cg.renderTo(svg);
-    cgNode = svg.select(".component-group").node();
-    c1Node = svg.select(".component-1").node();
+      assert.strictEqual(cg2.components().length, 1,
+        "second group should have 1 component after movement");
 
-    assert.isNotNull(cgNode, "component group was added back to the DOM");
-    assert.isNotNull(c1Node, "componet 1 was also added back to the DOM");
+      assert.strictEqual(cg1.components().length, 0,
+        "first group should have no components after movement");
 
-    svg.remove();
-  });
+      assert.strictEqual(c.parent(), cg2,
+        "component's parent after movement should be the group 2"
+      );
 
-  it("destroy()s its Components when destroy()ed", () => {
-    let c1 = new Plottable.Component().addClass("component-1");
-    let c2 = new Plottable.Component().addClass("component-2");
-    let cg = new Plottable.Components.Group([c1, c2]);
+      svg.remove();
+    });
 
-    let svg = TestMethods.generateSVG(200, 200);
-    cg.renderTo(svg);
+    it("destroy()s its Components when destroy()ed", () => {
+      let c1 = new Plottable.Component();
+      let c2 = new Plottable.Component();
+      let cg = new Plottable.Components.Group([c1, c2]);
+
+      let svg = TestMethods.generateSVG(200, 200);
+      cg.renderTo(svg);
 
-    cg.destroy();
-    assert.throws(() => c1.renderTo(svg), Error);
-    assert.throws(() => c2.renderTo(svg), Error);
+      cg.destroy();
+      assert.throws(() => c1.renderTo(svg), Error);
+      assert.throws(() => c2.renderTo(svg), Error);
 
-    svg.remove();
+      svg.remove();
+    });
   });
 
-  describe("requests space based on contents, but occupies total offered space", () => {
+  describe("Layout", () => {
     let SVG_WIDTH = 400;
     let SVG_HEIGHT = 400;
-
-    it("with no Components", () => {
+    it("requests no space when empty, but occupies all offered space", () => {
       let svg = TestMethods.generateSVG();
```

Comment:
Can't comment on the lines below, but all tests assume the `TestMethods.generateSVG()` defaults to 400 x 400. Specify arguments here.